pkg/executor/executortype/container/containermgr_test.go

```diff
@@ -0,0 +1,204 @@
+	os.Setenv("DEBUG_ENV", "true")
+	logger := loggerfactory.GetLogger()
+	kubernetesClient := fake.NewSimpleClientset()
+	fissionClient := fClient.NewSimpleClientset()
+	factory := make(map[string]genInformer.SharedInformerFactory, 0)
+	factory[metav1.NamespaceAll] = genInformer.NewSharedInformerFactory(fissionClient, time.Minute*30)
+
+	executorLabel, err := utils.GetInformerLabelByExecutor(fv1.ExecutorTypeContainer)
+	if err != nil {
+		t.Fatalf("Error creating labels for informer: %s", err)
+	}
+	cmInformerFactory := utils.GetInformerFactoryByExecutor(kubernetesClient, executorLabel, time.Minute*30)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	executor, err := MakeContainer(ctx, logger, fissionClient, kubernetesClient, "test",
+		factory, cmInformerFactory)
+	if err != nil {
+		t.Fatalf("container manager creation failed: %s", err)
+	}
+
+	ncm := executor.(*Container)
+
+	nsResolver := utils.NamespaceResolver{
+		FunctionNamespace: functionNamespace,
+		BuiderNamespace:   builderNamespace,
+		DefaultNamespace:  defaultNamespace,
+	}
+	ncm.nsResolver = &nsResolver
+
+	go ncm.Run(ctx)
+	t.Log("Container manager started")
+
+	for _, f := range factory {
+		f.Start(ctx.Done())
+	}
+	for _, informerFactory := range cmInformerFactory {
+		informerFactory.Start(ctx.Done())
+	}
+
+	t.Log("Informers required for container manager started")
+
+	waitSynced := make([]k8sCache.InformerSynced, 0)
+	for _, deplListerSynced := range ncm.deplListerSynced {
+		waitSynced = append(waitSynced, deplListerSynced)
+	}
+	for _, svcListerSynced := range ncm.svcListerSynced {
+		waitSynced = append(waitSynced, svcListerSynced)
+	}
+
+	if ok := k8sCache.WaitForCacheSync(ctx.Done(), waitSynced...); !ok {
+		t.Fatal("Timed out waiting for caches to sync")
+	}
+
+	funcUID, err := uuid.NewV4()
+	if err != nil {
+		t.Fatal(err)
+	}
+	funcSpec := fv1.Function{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      functionName,
+			Namespace: defaultNamespace,
+			UID:       types.UID(funcUID.String()),
+		},
+		Spec: fv1.FunctionSpec{
+			Environment: fv1.EnvironmentReference{
+				Name:      "",
+				Namespace: "",
+			},
+			InvokeStrategy: fv1.InvokeStrategy{
+				ExecutionStrategy: fv1.ExecutionStrategy{
+					ExecutorType: fv1.ExecutorTypeContainer,
+				},
+			},
+		},
+	}
+	_, err = fissionClient.CoreV1().Functions(defaultNamespace).Create(ctx, &funcSpec, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("creating function failed : %s", err)
+	}
+
+	funcRes, err := fissionClient.CoreV1().Functions(defaultNamespace).Get(ctx, functionName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error getting function: %s", err)
+	}
+	assert.Equal(t, funcRes.ObjectMeta.Name, functionName)
+
+	ctx2, cancel2 := context.WithCancel(context.Background())
+
+	wait.Until(func() {
+		t.Log("Checking for deployment")
+		ret, err := kubernetesClient.AppsV1().Deployments(functionNamespace).List(ctx2, metav1.ListOptions{})
```

Comment:
If the change made in the `containermgr.go` is reasonable to solve the issue, I add the tests for testing the functionality of func `RefreshFuncPods`. The tests are similar to what was added in the newdeploymgr_test.go.

But the tests are not passing. There are 2 failures

1. When we check for deployment in the `fission-function` namespace after creating the container function in the default namespace, it is not able to find the deployment and logs `Checking for deployment` constantly. I am not able 
to understand why is that the case since the function is created successfully.

2. Second error is also related to this here: https://github.com/NikhilSharmaWe/fission/blob/1c685c72909be12a9883fedb5b063837feb8eaaa/pkg/executor/executortype/container/containermgr_test.go#L176. That is to say that there are 0 deployments in the fission-function instead of 1.